src/core/credential.h

```diff
@@ -0,0 +1,45 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
```

Comment:
I think this .c/.h file should really be called `exec-credential.[ch]`, given it's still pretty closely related to execute.[ch], i.e. uses the Exec/exec_ prefix for most its definitions.